hooks/useOnClickAway.ts

```diff
@@ -0,0 +1,20 @@
+import { useEffect, RefObject } from 'react';
+
+const useOnClickAway = (ref: RefObject<Node>, action: () => void): void => {
+  useEffect(() => {
+    const handleAction = (event: MouseEvent | TouchEvent) => {
+      if (!ref.current || ref.current.contains(event.target as Node)) return;
```

Comment:
We can simply check if `event.target` exists instead of asserting its type `as Node`.